File: Point/point.cpp
#include <iostream>
#include <cmath>

#include "Point.h"

using namespace std;

Point::Point(double x, double y) {
    this->x = x;
    this->y = y;
} 

Point::~Point() {}

double Point::getX() const {
    return x;
}

double Point::getY() const {
    return y;
}

void Point::setX(double x) {
    this->x = x;
}

void Point::setY(double y) {
    this->y = y;
}

double Point::distance(const Point& other) const {
    return hypot(other.x-x, other.y-y);
}

bool Point::operator==(const Point& other) const {
    return x == other.x && y == other.y;
}

bool Point::operator!=(const Point& other) const {
    return !(*this == other);
}

ostream& operator<<(ostream& out, const Point& other) {
    return out << '(' << other.getX() << ", " << other.getY() << ')' << endl;
}

File: Java/src/com/gymfox/bcwc5/point/Point.java
/**
 * Created by andrew on 18.11.16.
 */
public class Point implements Cloneable{
    private double x;
    private double y;

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Point() {
        this(0.0, 0.0);
    }

    public void setX(double x) {
        this.x = x;
    }

    public void setY(double y) {
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public Point clone() throws CloneNotSupportedException {
        return (Point) super.clone();
    }

    public boolean equals(Point other) {
        return x == other.x && y == other.y;
    }

    public double distance(Point p) {
        return Math.hypot(x - p.x, y - p.y);
    }

    public String toString() {
        return "(" + x + "," + y + ")";
    }
}
